WcaOnRails/app/controllers/team_members_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+class TeamMembersController < ApplicationController
+  before_action :authenticate_user!
+  before_action -> { redirect_unless_user(:can_manage_committees?) }, only: [:new, :create]
+  before_action -> { redirect_unless_user(:can_edit_team?, team_from_params) }, only: [:edit]
+  before_action -> { redirect_unless_user(:can_edit_team?, team_from_params) }, only: [:update]
+
+  def new
+    @committee = committee_from_params
+    @team = team_from_params
+    @team_member = @team.team_members.new
+  end
+
+  def create
+    @committee = committee_from_params
+    @team = team_from_params
+    @team_member = @team.team_members.new(team_member_params_create)
+    if @team_member.save
+      flash[:success] = "Added user as a team member of team:" + @team.name
+      redirect_to committee_path(@committee)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @committee = committee_from_params
+    @team = team_from_params
+    @team_member = team_member_from_params
+  end
+
+  def update
+    @committee = committee_from_params
+    @team = team_from_params
+    @team_member = team_member_from_params
+    if @team_member.update_attributes(team_member_params_update)
+      flash[:success] = "Updated team"
+      redirect_to committee_path(@committee)
+    else
+      render :edit
+    end
+  end
```

Comment:
I think it makes sense redirect to team(s) instead of committee(s) in all actions above.